src/pages/index.js

```diff
@@ -5,52 +5,348 @@ import PopupWithForm from "../components/PopupWithForm.js";
-  profileForm.reset();
-  editProfilePopup.close();
-};
+// ----------------------------
+
+// like button
+// Function to handle like click
+
+function findCardById(cardId) {
+  return initialCards.find((card) => card._id === cardId || card.id === cardId);
+}
+function handleLikeClick(cardData) {
+  if (cardData._isLiked) {
+    api
+      .removeLike(cardData._id)
+      .then((updatedCard) => {
+        cardData._isLiked = false; // Update local like status
+        updateCardLikes(cardData._id, updatedCard.likes); // Update UI with new like count
+      })
+      .catch((err) => console.log(err));
+  } else {
+    api
+      .addLike(cardData._id)
+      .then((updatedCard) => {
+        cardData.isLiked = true; // Update local like status
+        updateCardLikes(cardData._id, updatedCard.likes); // Update UI with new like count
+      })
+      .catch((err) => console.log(err));
+  }
+}
+
+// Function to update card likes in the UI
+// function updateCardLikes(cardId, likes) {
+//   const cardElement = document.querySelector(`.card[data-id="${cardId}"]`); // Make sure card elements have data-id attribute
+//   const likeCountElement = cardElement.querySelector(".card__like-count");
+//   likeCountElement.textContent = likes.length;
+
+//   const likeButton = cardElement.querySelector(".card__button-like");
+//   if (likes.some((user) => user._id === currentUser._id)) {
+//     likeButton.classList.add("card__button-like_active");
+//   } else {
+//     likeButton.classList.remove("card__button-like_active");
+//   }
+// }
+
+// // Attach event listeners to like buttons
+// document.addEventListener("click", (event) => {
+//   if (event.target.classList.contains("card__button-like")) {
+//     const cardElement = event.target.closest(".card");
+//     const cardId = cardElement.dataset.id;
+//     const cardData = findCardById(cardId);
+//     console.log("Like button clicked for card ID:", cardId); // Debugging log
+//     console.log("Card data for like:", cardData);
+//     // const cardData = findCardById(cardId); // Implement this function to find the card object
+//     handleLikeClick(cardData);
+//   }
+// });
+
+// ----------------------------
+
+const profileSubmitButton = document.querySelector("#profile-submit-button");
+
+function handleProfileFormSubmit(event) {
+  // event.preventDefault();
+  profileSubmitButton.textContent = "Saving...";
+
+  const formData = new FormData(profileForm);
+
+  const formValues = {
+    name: formData.get("title"),
+    about: formData.get("description"),
+  };
+
+  api
+    .setUserInfo(formValues)
+    .then((data) => {
+      userInfo.setUserInfo({ name: data.name, job: data.about });
+      editProfilePopup.close();
+    })
+    .catch((error) => console.error("Error updating profile:", error))
+    .finally(() => {
+      profileSubmitButton.textContent = "Save";
+    });
+}
+
+profileForm.addEventListener("submit", handleProfileFormSubmit);
```

Comment:
Please, delete it

* Every form should be handled by  `PopupWithForm`

Reply:
i should delete this from index.js?

Reply:
> i should delete this from index.js?

yes